### PR TITLE
refactor(platform-server): minor cleanup in `insertEventRecordScript` logic

### PR DESCRIPTION
**Pull request by alan-agius4**
- Uses Set instead of a number of checks
- Remove unneeded `Array.from`

### CODE REVIEW

File: packages/platform-server/src/utils.ts [hunk 1]
```diff
@@ -133,49 +133,56 @@ function appendServerContextInfo(applicationRef: ApplicationRef) {
   });
 }
 
+const DISALLOWED_EVENT_TYPES = new Set<string>([
+  'mouseenter',
+  'mouseleave',
+  'pointerenter',
+  'pointerleave',
+]);
+const ALLOWED_CAPTAURE_EVENT_TYPES = new Set<string>(['focus', 'blur', 'error', 'load', 'toggle']);
```
Comment (eneajaho) [inline packages/platform-server/src/utils.ts:142]:
```suggestion
const ALLOWED_CAPTURE_EVENT_TYPES = new Set<string>(['focus', 'blur', 'error', 'load', 'toggle']);
```

File: packages/platform-server/src/utils.ts [hunk 2]
```diff
@@ -133,49 +133,56 @@ function appendServerContextInfo(applicationRef: ApplicationRef) {
...
+    return;
+  }
+
+  const captureEventTypes = [];
+  const eventTypes = [];
+  for (const eventType of eventTypesToBeReplayed) {
+    if (DISALLOWED_EVENT_TYPES.has(eventType)) {
+      continue;
+    }
+    if (ALLOWED_CAPTAURE_EVENT_TYPES.has(eventType)) {
```
Comment (eneajaho) [inline packages/platform-server/src/utils.ts:161]:
```suggestion
    if (ALLOWED_CAPTURE_EVENT_TYPES.has(eventType)) {
```

File: packages/platform-server/src/utils.ts [hunk 3]
```diff
@@ -133,49 +133,56 @@ function appendServerContextInfo(applicationRef: ApplicationRef) {
   });
 }
 
+const DISALLOWED_EVENT_TYPES = new Set<string>([
```
Comment (AndrewKushnir) [inline packages/platform-server/src/utils.ts:136]: It looks like those consts would intersect with a similar ones from PR #55799.

File: packages/platform-server/src/utils.ts [hunk 4]
```diff
@@ -133,49 +133,56 @@ function appendServerContextInfo(applicationRef: ApplicationRef) {
...
+    'document.body',
+    JSON.stringify(appId),
+    JSON.stringify(eventTypes),
+  ];
 
-    // Insert replay script right after inlined event dispatch script, since it
-    // relies on `__jsaction_bootstrap` to be defined in the global scope.
-    eventDispatchScript.after(replayScript);
+  if (captureEventTypes.length) {
+    jsActionBootstrapParams.push(JSON.stringify(captureEventTypes));
```
Comment (AndrewKushnir) [inline packages/platform-server/src/utils.ts:177]: nit:
```suggestion
  if (captureEventTypes.length > 0) {
    jsActionBootstrapParams.push(JSON.stringify(captureEventTypes));
```

**Pull request closed without merging**